### PR TITLE
Integrate Foster/Admin Application Page

### LINKED ISSUE #60: Fill Each Stage of Application Process With the Appropriate Component

### PR DESCRIPTION
**Pull request by petabite**
### Administrative Info
[GitHub Issue #58]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
- integrate application page components for foster/admin views
- add Foster Resources and Foster Matches sections for step 4 and 6 on foster view

### Testing
- manually changed the `role` variable in Application/index.js and ensured the view was correct

### Confirmation of Change 

**Foster Matches**
image

**Foster Resources**
image

### CODE REVIEW

File: client/src/pages/Application/index.js [hunk 1]
```diff
@@ -84,59 +82,26 @@ function Application() {
   const navigate = useNavigate();
   const [currentStep, setCurrentStep] = React.useState(0);
   const [currentSubStep, setCurrentSubStep] = React.useState("content");
+  // TODO: Use firebase user data
+  const role = "foster";
```
Comment (petabite) [inline client/src/pages/Application/index.js]: Set this to `"foster"` or `"admin"` to see either application view.

File: client/src/pages/Application/FosterView/Step4.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+    {
+      name: "tom",
+      imageRef: doggo,
+    },
+    {
+      name: "tom",
+      imageRef: doggo,
+    },
+  ]);
```
Comment (petabite) [inline client/src/pages/Application/FosterView/Step4.js:112]: Set this useState to an empty array (`[]`) to see the "Still looking for matches..." state on the Foster Matches component.

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Looking good Philip! Here's what I noticed:

1. The border width for most of the components is pretty inconsistent. I know Jacob and others have worked on this as well but can you try to make it more consistent across stages? It's mainly this Meetings component:
Screen Shot 2022-05-26 at 5 28 00 PM

Other components look like this:
Screen Shot 2022-05-26 at 5 28 25 PM

2. For the "Interview Confirmed" component on the foster side, should it have the interview location as well? Same thing for this component in the home screen and meet & greet stages.
Your code:
Screen Shot 2022-05-26 at 5 09 10 PM

Design:
Screen Shot 2022-05-26 at 5 10 12 PM

3.  For the first step on the admin side, it should display this loading component instead of what you have right now. You actually have this under step 2, so just move it to step 1.
Screen Shot 2022-05-26 at 5 20 51 PM

4. For step 6 on admin side, I know we don't have the Foster Profile component yet, but can you add the Dog Profile component there?
5. For all these components, can you add some padding below them so that they look like this? Maybe you just need to modify DefaultBody for this.
Screen Shot 2022-05-26 at 5 29 03 PM

File: client/src/pages/Application/index.js [hunk 3]
```diff
@@ -84,67 +84,37 @@ function Application() {
   const navigate = useNavigate();
   const [currentStep, setCurrentStep] = React.useState(0);
   const [currentSubStep, setCurrentSubStep] = React.useState("content");
+  // TODO: Use firebase user data
+  const [applicationView, setApplicationView] = React.useState("admin");
+  // TODO: Grab real application data from db
+  const [applicationState, setApplicationState] = React.useState({ status: "" });
```
Comment (petabite) [inline client/src/pages/Application/index.js]: Set `status` to `"rejected"` to see application rejected state

**Review by parth4apple — CHANGES_REQUESTED** [review]
Pretty awesome PR Philip. Here's some things I noticed.

Foster View:
1. On taller viewports, this component looks a little bit streched. Maybe limit the height of the container to the height of the image for non-mobile viewports.
Screen Shot 2022-05-30 at 2 39 39 PM
 
2. Similar issue with this component, the status update component on step 4, and foster resources component on step 6. The container div expands beyond what it needs to so it looks a little bit funky. Not a big deal if the css is too hard to fix.
Screen Shot 2022-05-30 at 2 41 27 PM
 
Screen Shot 2022-05-30 at 2 43 36 PM

Screen Shot 2022-05-30 at 2 45 46 PM

3. When you click adopt in step 6, it says "we are so excited for you to become a foster fail!". Maybe replace "fail" with "success". I think this might be a mistake in the figma.

Screen Shot 2022-05-30 at 2 49 51 PM
 
4. The content overflows on some ipad viewports for this component. I think it's because the buttons are statically sized.
Screen Shot 2022-05-30 at 3 03 14 PM


Admin View:
1. If there was a blurred background on the pass fail component or even just a black border if a blur is too hard to do, it would look a lot cleaner. 

Screen Shot 2022-05-30 at 2 59 53 PM

**Comment by parth4apple** [comment]
Hey Phillip, a lot better, but there are still some things to clean up.
Foster:

1. Follow up on number 4 in my previous code review, now all background div also overflows the page.
Screen Shot 2022-05-31 at 12 17 51 AM

2. The message area gets too small in tablet viewports and the dog image looks too big. Try to have a  similar scaling behavior to the interview scheduling component, I think that looks pretty nice. This is minor so if it's too tedious to fix, then don't worry about it. 

Screen Shot 2022-05-31 at 12 19 29 AM

 3. Add a little padding top.
Screen Shot 2022-05-31 at 12 23 16 AM

4. Minor thing - the titles "status update" and "foster resources" are very small on tablet/mobile views (stage 4 content, stage 6 content).

Screen Shot 2022-05-31 at 12 36 33 AM

Admin:
1. Another minor thing - make the Interview info and after interview titles inline with each other. 
Screen Shot 2022-05-31 at 12 28 06 AM

2. I think you refactoring the status update component to use flexbox modified the behavior of this page. The bottoms of each component do not match. If you add a little bit more whitespace to the bottom of the status update container in the meet and greet component, I think this should be fixed.

Screen Shot 2022-05-31 at 12 30 06 AM

**Review by parth4apple — APPROVED** [review]
LGTM

**Review by dnanjunda — APPROVED** [review]
LGTM! Nice work!

**Pull request merged**